urld829b80034_wjch111_PTSP2013Wang_tgz-pJ8-controllers_momcts_momctsCore_TestPlannerJ8/src/framework/utils/File2String.java

```diff
@@ -0,0 +1,165 @@
+package framework.utils;
+
+import java.io.*;
+import java.util.StringTokenizer;
+import java.util.Vector;
+
+/**
+ * Class to manage writing and reading from files.
+ * PTSP-Competition
+ * Created by Diego Perez, University of Essex.
+ * Date: 19/12/11
+ */
+public class File2String
+{
+    /**
+     * Returns a String with the contents of the file
+     * @param filename file name.
+     * @return a String with the contents of the file
+     */
+    public static String get(String filename) {
+      try {
+        File file = new File(filename);
+        byte[] b = new byte[(int) file.length()];
+        DataInputStream dis =
+          new DataInputStream(new FileInputStream(file));
```

Comment:
Recommendation generated by Amazon CodeGuru Reviewer. Leave feedback on this recommendation by replying to the comment or by reacting to the comment using emoji.

**Problem**
This line of code might contain a resource leak. Resource leaks can cause your system to slow down or crash.

**Fix**
Consider closing the resource created by the following constructor: *FileInputStream*. Execution paths that do not contain closure statements were detected. Ensure the object returned by *FileInputStream()* is closed in a try-finally block or declared in a try-with-resources block.

**More info**
[View resource management guidelines at oracle.com](https://www.oracle.com/technetwork/articles/java/trywithresources-401775.html) (external link).